apps/api_web/lib/api_web/controllers/portal/user_controller.ex

```diff
@@ -20,18 +20,26 @@ defmodule ApiWeb.ClientPortal.UserController do
     |> render("new.html", changeset: changeset)
   end
 
-  def create(conn, %{"user" => user_params}) do
-    case ApiAccounts.register_user(user_params) do
-      {:ok, user} ->
-        conn
-        |> put_session(:user_id, user.id)
-        |> configure_session(renew: true)
-        |> redirect(to: portal_path(conn, :index))
-
-      {:error, %ApiAccounts.Changeset{} = changeset} ->
+  def create(conn, %{"user" => user_params, "g-recaptcha-response" => recaptcha}) do
+    case Recaptcha.verify(recaptcha) do
```

Comment:
suggestion: this is probably clearer (and definitely less nested) as a `with`:
```elixir
with {:ok, _} <- Recaptcha.verify(recapcha),
  {:ok, user} <- ApiAccounts.register_user(user_params) do
    # good path
else
  {:error, %ApiAccounts.ChangeSet{} = changeset} ->
    # changeset error
  {:error, error} ->
    # log error
    # redirect user
end
```

Reply:
@paulswartz All about it, probably just my Python-brain 😂

Reply:
Updated - https://github.com/mbta/api/commit/5f443347a834bda7c27c87f8afb4673578dcdf0c